src/Runner.ts

```diff
@@ -0,0 +1,148 @@
+import * as program from "commander";
+import chalk from "chalk";
+
+import { getSupportedLanguages, setupUnibeautify } from "./index";
+import { BeautifyData } from "unibeautify";
+
+export class Runner {
+  public beautify(programArgs: IArgs): Promise<void> {
+    const {
+      // args: files,
+      language,
+      // outFile,
+      // replace,
+      // configFile,
+      configJson,
+      filePath,
+    } = programArgs;
+    return setupUnibeautify().then(unibeautify => {
+      if (!language) {
+        this.writeError("A language is required.");
+        return this.exit(1);
+      }
+      const config = this.parseConfig(configJson);
+      if (this.isTerminal) {
+        this.writeError("Beautify files is not yet supported.");
+        return this.exit(1);
+      } else {
+        return this.readFromStdin().then(text => {
+          const data: BeautifyData = {
+            filePath: filePath,
+            languageName: language,
+            options: config as any,
+            text,
+          };
+          return unibeautify
+            .beautify(data)
+            .then((result: string) => {
+              this.writeOut(result);
+              return this.exit(0);
+            })
+            .catch((error: Error) => {
+              this.writeError(error.message);
+              return this.exit(1);
+            });
+        });
+      }
+    });
+  }
+
+  public support(options: { json?: boolean; languages?: boolean }): void {
+    const printer: (info: SupportInfo) => void = options.json
+      ? this.jsonPrinter
+      : this.listPrinter;
+    const info: SupportInfo = {};
+    if (options.languages) {
+      info["languages"] = getSupportedLanguages();
+    }
+    // if (cmd.beautifiers) {
+    //   console.log("Coming soon!");
+    //   info["beautifiers"] = [];
+    // }
+    if (Object.keys(info).length === 0) {
+      // tslint:disable-next-line:no-console
+      this.writeError("Required option --languages is missing.");
```

Comment:
Why is `--languages` required for the list command?  I was thinking:

* If `--languages` or `--beautifiers` is not passed into the list command, just print both
or
* Print an error stating that one or both should be passed in

Reply:
Sure. I only did this because there was previously:

```javascript
console.error("Unrecognized option");
process.exit(1);
```

Reply:
Oh that was for anything other than `--languages` or `--beautifiers` being passed in.  e.g. `unibeautify list -r` would show "Unrecognized option".